src/bin/kafka_producer.rs

```diff
@@ -11,24 +24,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     tracing::subscriber::set_global_default(subscriber).expect("setting default subscriber failed");
 
-    let args: Vec<String> = env::args().collect();
+    let args = Cli::parse();
     let mut date = chrono::Utc::now().format("%Y%m%d").to_string();
     let mut limit = 0;
-    if args.len() > 1 {
-        let arg = &args[1];
-        if arg.len() == 8 {
-            info!("Using date from argument: {}", arg);
-            date = arg.to_string();
+
+    if let Some(d) = args.date {
+        if d.len() == 8 {
```

Comment:
Not critical for this PR, but I feel like there's a way to have clap parse the given date arg as a `chrono::DateTime`, and then this error case would be handled automatically by clap.